File: src/main/java/com/pickme/calendar/service/mapper/CalendarMapper.java
package com.pickme.calendar.service.mapper;

import java.util.List;

import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;

import com.pickme.calendar.dto.request.GetInterviewDto;
import com.pickme.calendar.dto.request.PostInterviewDto;
import com.pickme.calendar.dto.request.PutInterviewDto;
import com.pickme.calendar.dto.response.CalendarDto;
import com.pickme.calendar.entity.Calendar;
import com.pickme.calendar.entity.InterviewDetail;

@Mapper(componentModel = "spring") // Spring Bean으로 등록
public interface CalendarMapper {

	CalendarDto toDto(Calendar calendar);

	List<GetInterviewDto> toDto(List<InterviewDetail> interviewDetails);

	GetInterviewDto toDto(InterviewDetail interviewDetail);

	InterviewDetail toEntity(PostInterviewDto postInterviewDto);

	void toEntity(PutInterviewDto putInterviewDto, @MappingTarget InterviewDetail interviewDetail);
}

File: src/main/kotlin/com/pickme/calendar/adapter/inbound/web/mapper/CalendarMapper.kt
package com.pickme.calendar.adapter.inbound.web.mapper

import com.pickme.calendar.adapter.inbound.web.dto.request.GetScheduleDto
import com.pickme.calendar.adapter.inbound.web.dto.request.PostScheduleDto
import com.pickme.calendar.adapter.inbound.web.dto.request.PutScheduleDto
import com.pickme.calendar.adapter.inbound.web.dto.request.payload.CompanyDto
import com.pickme.calendar.adapter.inbound.web.dto.response.CalendarDto
import com.pickme.calendar.domain.model.Calendar
import com.pickme.calendar.domain.model.InterviewSchedule
import com.pickme.calendar.domain.model.InterviewUpdateSpec
import org.mapstruct.Mapper
import org.mapstruct.ObjectFactory

@Mapper(componentModel = "spring") // Spring Bean으로 등록
interface CalendarMapper {
    fun toDto(calendar: Calendar): CalendarDto

    fun toDto(interviewSchedules: List<InterviewSchedule>): List<GetScheduleDto>

    fun toDto(interviewSchedule: InterviewSchedule): GetScheduleDto

    fun toEntity(dto: CompanyDto): InterviewSchedule.Company

    fun toEntity(dto: PutScheduleDto): InterviewUpdateSpec

    @ObjectFactory
    fun toEntity(dto: PostScheduleDto): InterviewSchedule {
        return InterviewSchedule(
            company = toEntity(dto.company),
            date = dto.date,
            position = dto.position,
            category = dto.category,
            description = dto.description
        )
    }
}
